deploy.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# deploy.sh - Deployment script for News Search API and UI
+
+# Environment variables
+
+IMAGE_TAG="latest"  # Change this based on deployment (staging, production, v1.0)
+INDEXES="mc_search"
+ESHOSTS="http://ramos.angwin:9200,http://woodward.angwin:9200,http://bradley.angwin:9200"
+ESOPTS="{'timeout': 60, 'max_retries': 3}" # 'timeout' parameter is deprecated
+ELASTICSEARCH_INDEX_NAME_PREFIX="mc_search-*"
+TERMFIELDS="article_title,text_content"
+TERMAGGRS="top,significant,rare"
+APP_NAME="news-search-api"
+
+# Check if running as root
+if [[ $UID -ne 0 ]]; then
+    echo "This script must be run as root."
+    exit 1
+fi
+
+LOGIN_USER=$(whoami)
+if [ "x$LOGIN_USER" = x ]; then
+    # XXX fall back to whoami (look by uid)
+    echo could not find login user 1>&2
+    exit 1
+fi
+
+run_as_login_user() {
+    if [ $(whoami) = root ]; then
```

Comment:
Please be consistent in how you test for the root user.
I prefer to write in strict POSIX "sh" compatible dialect, ESPECIALLY for things that can be run as root.  bash has had security problems in the past, and I can EASILY imagine will in the future as well.